reports-scheduler/src/main/java/com/amazon/opendistroforelasticsearch/reportsscheduler/rest/ReportsSchedulerRestHandler.java

```diff
@@ -0,0 +1,161 @@
+    public static final String SCHEDULER_URI = "/_opendistro/reports_scheduler/schedule";
+
+    @Override
+    public String getName() {
+        return "Reports Scheduler Handler";
+    }
+
+    @Override
+    public List<Route> routes() {
+        return Collections.unmodifiableList(Arrays.asList(
+                new Route(RestRequest.Method.POST, SCHEDULER_URI),
+                new Route(RestRequest.Method.DELETE, SCHEDULER_URI)
+        ));
+    }
+
+    @Override
+    protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) {
+        if (request.method().equals(RestRequest.Method.POST)) {
+            // compose JobParameter object from request
+            String id = request.param("id");
+
+            //TODO: add start_time and Time_unit from request params or request body
+
+            //TODO: parse the content not params.
+            /**
+             * {
+             *
+             *     "name": "Flights monthly report",
+             *     "enabled": true, //boolean, default=true, required by ScheduledJobParameter
+             *     "schedule": {
+             *         "interval": {
+             *             "period": 1,
+             *             "unit": "MINUTES",
+             *             "start_time": 1553112384
+             *         }
+             *
+             *       },
+             *     "report_definition_id":"23dasdfdasdas"
+             *  }
+             */
+
+            if(id == null) {
+                throw new IllegalArgumentException("Must specify id");
+            }
+
+            IndexRequest indexRequest = new IndexRequest()
+                    .index(ReportsSchedulerPlugin.JOB_INDEX_NAME)
+                    .id(id)
+                    .source(request.requiredContent(), XContentType.JSON);
+
+            return restChannel -> {
+                // index the job parameter
+                client.index(indexRequest, new ActionListener<IndexResponse>() {
+                    @Override
+                    public void onResponse(IndexResponse indexResponse) {
+                        try {
+                            RestResponse restResponse = new BytesRestResponse(RestStatus.OK,
+                                    indexResponse.toXContent(JsonXContent.contentBuilder(), null));
+                            restChannel.sendResponse(restResponse);
+                        } catch(IOException e) {
+                            restChannel.sendResponse(new BytesRestResponse(RestStatus.INTERNAL_SERVER_ERROR, e.getMessage()));
+                        }
+                    }
+
+                    @Override
+                    public void onFailure(Exception e) {
+                        restChannel.sendResponse(new BytesRestResponse(RestStatus.INTERNAL_SERVER_ERROR, e.getMessage()));
+                    }
+                });
+            };
+        } else if (request.method().equals(RestRequest.Method.DELETE)) {
```

Comment:
may be making use of POST/DELETE handlers instead of if/else would help unit test easily?